database/sql/V011__create_settlement_version.sql

```diff
@@ -0,0 +1,9 @@
+CREATE TABLE settlement_version
```

Comment:
What's the advantage of using a separate table for this? Shouldn't contract just be a column on the order table? Also I wouldn't populate the contract address values hardcoded (except for migrations) but instead have code read the current address from the used contract artefact in the rust code and use that address. This way we depend on this string only in one place.

Reply:
I have similar thoughts. Having an integer version (or an enum version) and mapping it to contract address in Rust code will give us more flexibility IMO. This is where https://github.com/gnosis/ethcontract-rs/issues/545 would've been useful.

Reply:
One advantage of having the contract address directly in the DB is that it is implicitly included in the signature. Specifically, if we were to redeploy our contracts (with a different salt, so they get a different deterministic address), then the current orders and signatures would no longer be valid, even if the code is identical, because the contract address is used for deriving the domain separator which is included in the message that is used for signing.

I think manually maintaining versions in Rust would be more error prone than having the orderbook API programmatically include the settlement contract address that it is using when inserting new rows.

Reply:
I am getting the impression that more of us are in favour of putting the `settlement_version`(contract address) directly in the orders table. This is totally fine by me as long as we are all comfortable with the redundant bytes (occupying ~1 GB per 50k orders).

Thumbs up 👍🏽 for eliminating the settlement version table and putting the address directly in the orders table.
Thumbs down 👎🏽 for keeping it as is.

Reply:
Just as a general comment, I think subjective polls (with thumb up/down votes) are good for naming questions, but technical decisions like this should be guided by technical arguments and not by a majority vote (and ideally we would at the end of the discussion agree what is the better solutions, because the arguments make sense).

Reply:
True. Well, now that I see the occupancy calculation was off by a factor of 1000, I think the better option is to eliminate this table and put the address directly in the orders table.

Reply:
I'd still be curious to see if there is a performance impact for including `bytea` in tables vs using a separate table with FK. I don't think it matters for this PR though and would be in favour of the simpler option (including the `settlment_contract` address directly in the order metadata table) and then circling back to this once we can actually measure the performance impact of a change. an additional 20 bytes per order seems a relatively small price to pay (50 * 20 = 1000 == 1KB, I'm not sure where the 1MB comes from).